tests/test_fastapi_json_schema.py

```diff
@@ -0,0 +1,125 @@
+from pydantic.errors import PydanticInvalidForJsonSchema
+from pydantic.json_schema import GenerateJsonSchema, JsonSchemaValue
+
+
+class _ErrorKey(str):
+    pass
+
+
+class FastAPIGenerateJsonSchema(GenerateJsonSchema):
+    """
+    Idea: This class would be exported from FastAPI, and if users want to modify the way JSON schema is generated
+    in FastAPI, they should inherit from it and override it as appropriate.
+
+    In the JSON schema generation logic, FastAPI _could_ also attempt to work with classes that inherit directly from
+    GenerateJsonSchema by doing something like:
+
+        if UserGenerateJsonSchema.handle_invalid_for_json_schema is GenerateJsonSchema.handle_invalid_for_json_schema:
+            # The method has not been overridden; inherit from FastAPIGenerateJsonSchema
+            UserGenerateJsonSchema = type(
+                "UserGenerateJsonSchema", (FastAPIGenerateJsonSchema, UserGenerateJsonSchema), {}
+            )
+        else:
+            raise TypeError(f"{UserGenerateJsonSchema.__name__} should inherit from FastAPIGenerateJsonSchema")
+
+    I'm not sure which approach is better.
+    """
+
+    def handle_invalid_for_json_schema(self, schema: CoreSchema | TypedDictField, error_info: str) -> JsonSchemaValue:
+        # NOTE: I think it may be a good idea to rework this method to either not use CoreMetadataHandler,
+        #    and/or to make CoreMetadataHandler a public API.
+        if CoreMetadataHandler(schema).get_modify_js_function():
+            # Since there is a json schema modify function, assume that this type is meant to be handled,
+            # and the modify function will set all properties as appropriate
+            return {}
+        else:
+            error = PydanticInvalidForJsonSchema(f'Cannot generate a JsonSchema for {error_info}')
+            return {_ErrorKey('error'): error}
+
+
+@dataclass
+class ErrorDetails:
+    path: list[Any]
+    error: PydanticInvalidForJsonSchema
+
+
+def collect_errors(schema: JsonSchemaValue) -> list[ErrorDetails]:
+    errors: list[ErrorDetails] = []
+
+    def _collect_errors(schema: JsonSchemaValue, path: list[Any]) -> None:
+        if isinstance(schema, dict):
+            for k, v in schema.items():
+                if isinstance(k, _ErrorKey):
+                    errors.append(ErrorDetails(path, schema[k]))
+                _collect_errors(v, list(path) + [k])
+        elif isinstance(schema, list):
+            for i, v in enumerate(schema):
+                _collect_errors(v, list(path) + [i])
+
+    _collect_errors(schema, [])
+    return errors
+
+
+def test_inheritance_detection() -> None:
+    class GenerateJsonSchema2(GenerateJsonSchema):
+        pass
+
+    assert GenerateJsonSchema2.handle_invalid_for_json_schema is GenerateJsonSchema.handle_invalid_for_json_schema
+    assert (
+        FastAPIGenerateJsonSchema.handle_invalid_for_json_schema
+        is not GenerateJsonSchema.handle_invalid_for_json_schema
```

Comment:
this is just a quick proof that my suggestion above _does_ work for checking if a method was overridden.